docs/source/fakes.rst

```diff
@@ -371,3 +371,17 @@ Modifying the :code:`receive` function
 .. code-block:: typescript
 
   myFake.receive.returns();
+
+Delegated calls
```

Comment:
I would strongly suggest adding a paragraph or two around Diamonds, as it might not be obvious to everyone. Something like:

```
If you're using the diamond pattern ([EIP-2535](https://eips.ethereum.org/EIPS/eip-2535)), you can use Smock to ensure that certain facets of your diamond are being called by others.

To do so, you can simply perform a DiamondCut operation, replacing one of your facets by a fake or a mock. It could look like something like this in your setup:

```typescript
const sigsFromABI = (abi: any[]): string[] =>
  abi
    .filter(frag => frag.type === 'function')
    .map(frag => Interface.getSighash(FunctionFragment.from(frag)));


myFacetFake = await smock.fake('MyFacet');

const selectors = sigsFromABI((await artifacts.readArtifact('MyFacet')).abi);
const diamond = await ethers.getContract('OurDiamond');
await diamond.diamondCut([{
  facetAddress: myFacetFake.address,
  action: FacetCutAction.Replace,
  functionSelectors: selectors
}], ZERO_ADDRESS, '0x')
```

This replaces `MyFacet`'s address from your diamond proxy with the address of your fake facet.

Alternatively, you could also decide to directly to deploy your diamond without `MyFacet` and perform a DiamondCut specifying the `FacetCutAction.Add` instead of `FacetCutAction.Replace`.

Afterwards, testing that a given facet is being called by another becomes simple. For example if the facet you're testing has a `foo(string memory message)` method that in turns calls `MyFacet(address(this)).bar(message)`, you can test that `MyFacet.bar` is called:

```ts
        await diamond.foo('Whatever');
        expect(myFacetFake.bar).to.have.been
          .calledWith('Whatever)
          .delegatedFrom(diamond.address);
```

Reply:
Will do in a separate PR